### ISSUE TITLE
incorrect comparison brackets

### ISSUE THREAD

**Issue by tanj** [tango-controls/TangoSourceDistribution#9]
https://github.com/tango-controls/TangoSourceDistribution/blob/a155290d3d5661079ae4913a67ad8facd9c003f0/configure#L19576

https://github.com/tango-controls/TangoSourceDistribution/blob/a155290d3d5661079ae4913a67ad8facd9c003f0/configure#L19983

These comparisons are not doing what is expected and will always return false. You should be using double-brackets for these tests.

http://tldp.org/LDP/abs/html/comparison-ops.html

```
[[ $a == z* ]]   # True if $a starts with an "z" (pattern matching).
[[ $a == "z*" ]] # True if $a is equal to z* (literal matching).

[ $a == z* ]     # File globbing and word splitting take place.
[ "$a" == "z*" ] # True if $a is equal to z* (literal matching).

# Thanks, Stéphane Chazelas
```

**Comment by Ingvord** [comment]
@tanj , thanks for the suggestion!

This could be a reason for [this issue](https://github.com/tango-controls/TangoSourceDistribution/pull/8#issuecomment-425403756)

I will check that out...

**Comment by Ingvord** [comment]
Nope, that did not work:

./configure still fails on Debian stretch

```
configure: WARNING: Not supported mysql release. Should be 5.5.x or above. Please update !
checking that generated files are newer than configure... done
```
...
```
Configuration (tango):

        Source code location:   .
        Version:                9.3.2
        Compiler:               gcc,g++

        OMNIORB PATH:           /usr/local
        OMNIORB VERSION:        

        ZMQ PATH:               /usr
        ZMQ VERSION:            4.2.1

        JAVA PATH:              /usr/bin/java
        JAVA VERSION:           9-Debian

        MYSQL CLIENT LIB:   -lmysqlclient
        MYSQL VERSION:     
        MYSQL CONNECTION:  

build:
        libraries:              yes
        java application:       yes
        access control server:  no
        database server:        no
        database schema create: no

Please check whether the configuration I detected matches what you
would like to have.


ingvord@vb-stretch:~/Downloads/tango-9.3.2$ mysql
Welcome to the MariaDB monitor.  Commands end with ; or \g.
Your MariaDB connection id is 2
Server version: 10.1.26-MariaDB-0+deb9u1 Debian 9.1

Copyright (c) 2000, 2017, Oracle, MariaDB Corporation Ab and others.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

MariaDB [(none)]> exit
Bye
```

**Comment by tanj** [comment]
In order to get past the sql detection I had to have these options.

```
./configure --enable-mariadb --with-mariadbclient-include=/usr/include/mysql --with-mariadbclient-lib=/usr/lib64
```

I found that the mysql_version.h file is not in the /usr/include/mariadb path on a gentoo system. mariadb-10.1.34

**Comment by Ingvord** [comment]
Wow, @tanj , that is definitely a step forward. Meaning that it is not a problem of `./configure`. Thanks. 

Unfortunately this did not work out for on my Debian 9 VB:

```console
ingvord@vb-stretch:~/Downloads/tango-9.3.2$ ./configure --enable-mariadb --with-mariadbclient-include=/usr/include/mysql --with-mariadbclient-lib=/usr/lib/x86_64-linux-gnu
```
...
```
checking for mariadbclient ... checking for mariadb includes... 
found -I/usr/include/mysql
checking for mariadb libx... 
defaultd to -L/usr/lib/x86_64-linux-gnu
no
can't compile a simple app with mysql_connnect in it. bad.

***
*** mariadbclient test source had problems, check your config.log .
*** Also try one of the following switches :
***   --with-mariadbclient-prefix=PFX
***   --with-mariadbclient-include=DIR
***   --with-mariadbclient-lib=DIR
***
configure: WARNING: No libmariadbclient libs found
```
...
```
Configuration (tango):

        Source code location:   .
        Version:                9.3.2
        Compiler:               gcc,g++

        OMNIORB PATH:           /usr/local
        OMNIORB VERSION:        

        ZMQ PATH:               /usr
        ZMQ VERSION:            4.2.1

        JAVA PATH:              /usr/bin/java
        JAVA VERSION:           9-Debian

        MARIADB CLIENT LIB:  -L/usr/lib/x86_64-linux-gnu 
        MARIADB VERSION:     10.1.26-MariaDB
        MARIADB CONNECTION:  failed

build:
        libraries:              yes
        java application:       yes
        access control server:  no
        database server:        no
        database schema create: no

Please check whether the configuration I detected matches what you
would like to have.
```

while

```console
ingvord@vb-stretch:~/Downloads/tango-9.3.2$ ls -al /usr/lib/x86_64-linux-gnu/libmariadbclient.so
lrwxrwxrwx 1 root root 22 Aug 10  2017 /usr/lib/x86_64-linux-gnu/libmariadbclient.so -> libmariadbclient.so.18
```

**Comment by bourtemb** [comment]
The check for mariadbclient in the configure script seems to be obsolete.
It is trying to compile a simple app by linking with -lmariadb instead of -lmariadbclient or -lmariadbclient_r. (See configure script at line 19799):
`MARIADBCLIENT_LIBS="-lmariadb"`

It looks like all this section should be rewritten in the configure script.
I think the script should take advantage of pkg-config whenever possible. This should be possible for mariadb, depending on how the user installed mariadb and depending on the installed mariadb version.
Is there a standard place on the web where we could get the latest version of these kinds of checks?

**Comment by bourtemb** [comment]
Ooops, sorry, my mistake... I don't have much experience with autotools.
Forgot to regenerate the configure script with bootstrap...
But when I do that, I get the following errors:

```
configure.ac:459: error: required file 'doc/src/Makefile.in' not found
configure.ac:459: error: required file 'doc/src/ds_writing/Makefile.in' not found
configure.ac:459: error: required file 'doc/src/ds_model/Makefile.in' not found
configure.ac:459: error: required file 'doc/src/dance/Makefile.in' not found
configure.ac:459: error: required file 'doc/src/java_api/Makefile.in' not found
configure.ac:459: error: required file 'doc/src/java_api/picture/Makefile.in' not found
configure.ac:459: error: required file 'doc/src/gen_api/Makefile.in' not found
configure.ac:459: error: required file 'doc/src/advanced/Makefile.in' not found
cppserver/AbstractClass/AccessControl/Makefile.am: installing 'config/depcomp'
doc/Makefile.am:3: error: required directory doc/src does not exist
```

doc/src was needed to generate the tango book in the past since we needed to provide the source files needed to generate the documentation in order to be compliant with the Debian rules. Thanks @taurel for the history lesson.
I think we should replace this with the documentation sources which are now in tango-doc repository and provide the sphinx files.

We should not modify the configure script since this one is generated by autoconf. We should rather work on autoconf.ac

**Comment by Ingvord** [comment]
@bourtemb, 

> I think we should replace this with the documentation sources which are now in tango-doc repository and provide the sphinx files.

Already done in ant script, in #8, but slightly different - it downloads pdf file generated by readthedocs.

**Comment by bourtemb** [comment]
As I understood, we need to provide the sources used to generate this pdf if we want to be compliant with the Debian rules.

**Comment by bourtemb** [comment]
A potential solution for the original problem of this issue would be to add extra brackets on the following line:
https://github.com/tango-controls/TangoSourceDistribution/blob/7651ca83d7c6412ead17c9cb8db66e86560d8cce/assets/m4/mysql_release.m4#L22
which would be replaced with:
`if [[[ "$MARIA" == *aria* ]]]; then `

It looks like the problem comes from the fact that the external brackets are removed by the autotools when generating the configure script because the brackets are special characters in m4 macros as described in the autoconf documentation: https://www.gnu.org/software/autoconf/manual/autoconf-2.69/html_node/Active-Characters.html#Active-Characters

This might help... but we might need to update the macros as well since they haven't been updated since several years.
The tricky and time consuming part is to test many different installation configurations.

**Comment by t-b** [comment]
Closed via #49.
